### PR TITLE
Better lint messages for `avoid_renaming_method_parameters`

### PR DESCRIPTION
**Pull request by asashour**
Related to the message description of #2683 

Dart `fix` to be separately handled.

### CODE REVIEW

File: lib/src/rules/avoid_renaming_method_parameters.dart [hunk 1]
```diff
@@ -108,7 +112,8 @@ class _Visitor extends SimpleAstVisitor<void> {
       var paramIdentifier = parameters[i].identifier;
       if (paramIdentifier != null &&
           paramIdentifier.name != parentParameters[i].name) {
-        rule.reportLint(parameters[i].identifier);
+        rule._reportLintForNodeWithDescription(paramIdentifier,
+            "Don't rename parameter '${parentParameters[i].name}' of the overridden method to '${paramIdentifier.name}'.");
```
Comment (bwilkerson) [inline lib/src/rules/avoid_renaming_method_parameters.dart:116]: Consider as an alternative "The parameter '{0}' should have the name '{1}' to match the name used in the overridden method.".

File: lib/src/rules/avoid_renaming_method_parameters.dart [hunk 2]
```diff
@@ -62,10 +62,14 @@ class AvoidRenamingMethodParameters extends LintRule implements NodeLintRule {
     var visitor = _Visitor(this, context);
     registry.addMethodDeclaration(this, visitor);
   }
+
+  void _reportLintForNodeWithDescription(AstNode node, String description) {
+    reporter.reportErrorForNode(LintCode(name, description), node);
```
[the remark this answers is not in the document]
Comment (asashour) [reply lib/src/rules/avoid_renaming_method_parameters.dart]: Thanks for the hint, I guess I copied the original code from another lint.

A question remains, how to test the `correction` parameter of `LintCode`?

[the remark this answers is not in the document]
Comment (bwilkerson) [reply lib/src/rules/avoid_renaming_method_parameters.dart]: I'm not aware of any way to test that. @pq might know of a way.

Outside of the linter we don't test the specific problem or correction messages for _any_ of our diagnostics. We do have tests to ensure that if an `ErrorCode` (and `LintCode` is a subclass of `ErrorCode`) has a problem message and/or correction message that those messages end up being available to the user, but encoding specific messages makes it harder for the messages to be updated.

File: lib/src/rules/avoid_renaming_method_parameters.dart [hunk 3]
```diff
@@ -59,16 +59,21 @@ class AvoidRenamingMethodParameters extends LintRule implements NodeLintRule {
...
     registry.addMethodDeclaration(this, visitor);
   }
 }
 
 class _Visitor extends SimpleAstVisitor<void> {
+  static const LintCode parameterCode = LintCode(
+      "avoid_renaming_method_parameters",
+      "The overridden method should have same parameter names as the parent method.",
+      correction:
+          "The parameter '{0}' should have the name '{1}' to match the name used in the overridden method.");
```
Comment (bwilkerson) [inline lib/src/rules/avoid_renaming_method_parameters.dart:72]: The message you currently have as the correction message would be good as the problem message, but the correction message should take the form "Try <solution>.", so perhaps "Try renaming the parameter.".

**Review by bwilkerson — APPROVED** [review]

**Review by pq — APPROVED** [review]
Nice!

💎

**Pull request merged**